lib/Service.php

```diff
@@ -189,6 +189,29 @@ function write($rootElementName, $value, $contextUri = null) {
 
     }
 
+    /**
+     * Maps the given value-object to the given rootElementName within the given namespace.
+     *
+     * The $rootElement must be specified in clark notation.
+     *
+     * The $className refernces a regular php class which must provide a __construct() without arguments.
+     * This methods creates instances of this class which later on picks up all xml child elements as public properties.
+     *
+     * @param string $rootElementName
+     * @param object $className
+     * @param string $namespace
+     */
+    function mapValueObject($rootElementName, $className, $namespace) {
+        if (!class_exists($className)) {
+            throw new \InvalidArgumentException('class "' . $className . '" does not exist');
+        }
+        $this->elementMap['{' . $namespace . '}' . $rootElementName] = function(Reader $reader) use ($className, $namespace) {
```

Comment:
I dropped the `$this->namespaceMap[$namespace]=null` expression here, because it had side effects.
I lead to all elements of this namespaces beeing represented by its localName instead of clark-notated (also for this not "registered" via `mapValueObject`).

After dropping the line we are still green, so nothing important ;)